src/Interpreters/TreeOptimizer.cpp

```diff
@@ -171,30 +171,13 @@ void optimizeGroupBy(ASTSelectQuery * select_query, ContextPtr context)
 
             /// copy shared pointer to args in order to ensure lifetime
             auto args_ast = function->arguments;
-
-            if (select_query->group_by_with_rollup)
-            {
-                const auto insert_position = group_exprs.begin() + i;
-                group_exprs.erase(insert_position);
-                std::copy_if(
-                        std::begin(args_ast->children), std::end(args_ast->children),
-                        std::inserter(group_exprs, insert_position),
-                        [&is_literal](const ASTPtr & arg) { return !is_literal(arg); }
-                );
-            }
-            else
-            {
-                /** remove function call and take a step back to ensure
-                  * next iteration does not skip not yet processed data
-                  */
-                remove_expr_at_index(i);
-
-                /// copy non-literal arguments
-                std::remove_copy_if(
-                        std::begin(args_ast->children), std::end(args_ast->children),
-                        std::back_inserter(group_exprs), is_literal
-                );
-            }
+            /// Replace function call in 'group_exprs' with non-literal arguments.
+            const auto & insert_position = group_exprs.begin() + i;
+            group_exprs.erase(insert_position);
```

Comment:
tbh I know nothing about `absl::InlinedVector` implementation, but I still suppose that `insert_position` might became invalid after this call (if we reduced vector internal buffer). let's recalculate it again before l.179

Reply:
Ok